features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/domain/ResourceRequestBodyFactoryTest.kt

```diff
@@ -0,0 +1,235 @@
+import com.datadog.android.sessionreplay.internal.domain.ResourceRequestBodyFactory.Companion.MULTIPLE_APPLICATION_ID_ERROR
+import com.datadog.android.sessionreplay.internal.domain.ResourceRequestBodyFactory.Companion.NAME_IMAGE
+import com.datadog.android.sessionreplay.internal.domain.ResourceRequestBodyFactory.Companion.NAME_RESOURCE
+import com.datadog.android.sessionreplay.internal.domain.ResourceRequestBodyFactory.Companion.NO_RESOURCES_TO_SEND_ERROR
+import com.datadog.android.sessionreplay.internal.domain.ResourceRequestBodyFactory.Companion.TYPE_KEY
+import com.datadog.android.sessionreplay.internal.domain.ResourceRequestBodyFactory.Companion.TYPE_RESOURCE
+import com.datadog.android.sessionreplay.internal.exception.InvalidPayloadFormatException
+import com.datadog.android.utils.verifyLog
+import com.google.gson.JsonObject
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import okhttp3.MultipartBody
+import okhttp3.RequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import okio.Buffer
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(ForgeConfigurator::class)
+internal class ResourceRequestBodyFactoryTest {
+    private lateinit var testedRequestBodyFactory: ResourceRequestBodyFactory
+
+    @StringForgery
+    private lateinit var fakeApplicationId: String
+
+    @StringForgery
+    private lateinit var fakeFilename: String
+
+    @StringForgery
+    private lateinit var fakeImageRepresentation: String
+
+    @Mock
+    lateinit var mockInternalLogger: InternalLogger
+
+    private lateinit var fakeMetaData: JsonObject
+
+    @BeforeEach
+    fun `set up`() {
+        fakeMetaData = JsonObject()
+        fakeMetaData.addProperty(APPLICATION_ID_KEY, fakeApplicationId)
+        fakeMetaData.addProperty(FILENAME_KEY, fakeFilename)
+
+        testedRequestBodyFactory = ResourceRequestBodyFactory(mockInternalLogger)
+    }
+
+    @Test
+    fun `M return valid requestBody W create()`() {
+        // Given
+        val fakeRawBatchEvent = RawBatchEvent(
+            data = fakeImageRepresentation.toByteArray(),
+            metadata = fakeMetaData.toString().toByteArray(Charsets.UTF_8)
+        )
+        val fakeListResources = listOf(fakeRawBatchEvent)
```

Comment:
What prevents us from generating multiple items for this list? This will be real use-case scenario. If I understand it correctly, single batch may contain multiple images.